### PR TITLE
Fix background in login page

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Bring background image back as it disapeared during the scsscacher deletion
- Fix background-size

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

File: core/css/guest.css [hunk 1]
```diff
@@ -24,7 +24,7 @@ body {
 	color: var(--color-primary-text);
 	text-align: center;
 	background-color: var(--color-primary);
-	background-image: var(--image-background, var(--image-background-plain, url('../../../core/img/background.svg'), linear-gradient(40deg, #0082c9 0%, #30b6ff 100%)));
+	background-image: var(--image-background, url('../img/background.png')), linear-gradient(40deg, #0082c9 0%, #30b6ff 100%);
```
Comment (nickvergessen) [inline core/css/guest.css:27]: This gradient needs to be overwritten by the theming app.

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 2]
```diff
@@ -199,9 +199,9 @@ public function getCSSVariables(): array {
 				$variables["--image-background-plain"] = 'true';
 				continue;
 			} else if ($image === 'background') {
-				$variables['--image-background-size'] = 'cover';
+				$variables['--image-background-size'] = !$this->imageManager->hasImage($image) ? '275px, contain' : 'cover';
 			}
-			$variables["--image-$image"] = "url('".$this->imageManager->getImageUrl($image)."')";
+			$variables["--image-$image"] = "url('".$this->imageManager->getImageUrl($image, true, $this->primaryColor)."')";
```
Comment (skjnldsv) [inline apps/theming/lib/Themes/DefaultTheme.php:204]: Sorry, but no.
This is a weird feature. If I select my own image as a login, I do not want this weird overlay containing apps icons

File: core/css/guest.css [hunk 3]
```diff
@@ -24,7 +24,7 @@ body {
 	color: var(--color-primary-text);
 	text-align: center;
 	background-color: var(--color-primary);
-	background-image: var(--image-background, var(--image-background-plain, url('../../../core/img/background.svg'), linear-gradient(40deg, #0082c9 0%, #30b6ff 100%)));
+	background-image: var(--image-background, url('../img/background.png')), var(--gradient-primary-background);;
```
Comment (skjnldsv) [inline core/css/guest.css:27]: You're removing features here.
The login page has three options (by order of priority):
1. the user uploaded a custom background image
2. the user disabled the background image for primary only
3. the default Nextcloud background.svg
4. Nextcloud default linear gradient

With your changes, I cannot remove the background image anymore

If I revert ba3f5337f356ac51a1b1d012c09f3e4ed6c096e3 from https://github.com/nextcloud/server/pull/33452, this works again
Peek 16-08-2022 08-31

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: core/css/guest.css (hunk 3 above)
Comment (skjnldsv) [reply core/css/guest.css:27]: https://github.com/nextcloud/server/pull/33452#issuecomment-1216210947

**Comment by skjnldsv** [comment]
Fix in https://github.com/nextcloud/server/pull/33556
and https://github.com/nextcloud/firstrunwizard/pull/731

**Pull request closed without merging**